ScoutHelper/Utils/Functional/FunctionalExtensions.cs

```diff
@@ -24,6 +24,12 @@ public static class FunctionalExtensions {
 			)
 			.Select(result => result.Value);
 
+	public static IEnumerable<Maybe<U>> MaybeSelect<T, U>(this IEnumerable<Maybe<T>> source, Func<T, U> transform) =>
+		source.Select(maybeValue => maybeValue.Select(transform));
+
+	public static IEnumerable<Maybe<U>> MaybeSelectMany<T, U>(this IEnumerable<Maybe<T>> source, Func<T, Maybe<U>> transform) =>
```

Comment:
your `Collect` sounds a lot like c#'s `Select`, hmm. or am i misunderstanding?

the intent with these is that i have situations where i have a list of `Maybe`s and i want to do `maybe.Select(transform)` on each one. and that's just annoying because it looks like:
```csharp
enumerable.Select(maybe => maybe.Select(transform))
```
and that was annoying to type repeatedly, and make things harder to read.

the difference between `MaybeSelect` and `MaybeSelectMany` is that `SelectMany` takes a transform that returns the monad type (in this case `Maybe`). so `MaybeSelect(Func<T,U>)` vs `MaybeSelectMany(Func<T,Maybe<U>>)`. i hate c#'s naming but it's just the difference between `map` and `flatMap/bind` in other languages.

but your questions got me thinking that maybe the more clear naming here would be `SelectOverMaybe` because it's doing a `Select` but doing it to the underlying `Maybe`. and then i would have `SelectManyOverMaybe` hmm.